### PR TITLE
feat(mantine, table):  reset row selection if user click outside the table

### PR DESCRIPTION
**Pull request by aiheon**
### Proposed Changes

UITOOL-956

A comment was raised in a pr that the Mantine table was not unselecting row if the user click outside the table. 
Now it does 😄 

No demo in Plasma website here but here's a video from me hacking a bit the Badge demo page 😆 

https://user-images.githubusercontent.com/63734941/199560856-46f68b13-ff93-4585-95fe-73596043b911.mp4

### Potential Breaking Changes

None

### Acceptance Criteria

-   [X] The proposed changes are covered by unit tests
-   [X] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/mantine/src/components/table/Table.tsx [hunk 1]
```diff
@@ -175,6 +175,10 @@ export const Table: TableType = <T,>({
         onMount?.({...state, ...form.values});
     }, []);
 
+    const outSideClickRef = useClickOutside(() => {
+        table.resetRowSelection(true);
+    });
```
Comment (GermainBergeron) [inline packages/mantine/src/components/table/Table.tsx:180]: Does it behave correctly when you have a predicate or date picker in the table?

(I'll be happy when we'll have the demo 👀 )

Comment (aiheon) [reply packages/mantine/src/components/table/Table.tsx:180]: I can try it, I still have my setup in a stash 👍 
i'll make you a video 📦

Comment (aiheon) [reply packages/mantine/src/components/table/Table.tsx:180]: 🥲 Good call, I need to fix this 👍 

https://user-images.githubusercontent.com/63734941/200009231-ea122212-a442-4e84-a876-12abe5195f94.mp4

Comment (aiheon) [reply packages/mantine/src/components/table/Table.tsx:180]: Clicking in the table header is does not remove the selected row now

https://user-images.githubusercontent.com/63734941/200016827-e8197ab4-0742-43c8-b9f9-437832ed1872.mp4

**Review by toofff — APPROVED** [review]

**Review by FelixBlaisThon — APPROVED** [review]
looks good

**Pull request merged**